src/test/ui/issues/issue-73886.stderr

```diff
@@ -4,14 +4,19 @@ error[E0605]: non-primitive cast: `&&[i32; 1]` as `&[_]`
 LL |     let _ = &&[0] as &[_];
    |             ^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
 
-error[E0605]: non-primitive cast: `u32` as `Option<_>`
+error[E0620]: cast to unsized type: `u32` as `Option<_>`
   --> $DIR/issue-73886.rs:4:13
    |
 LL |     let _ = 7u32 as Option<_>;
-   |             ^^^^^^^^^^^^^^^^^ help: consider using the `From` trait instead: `Option<_>::from(7u32)`
+   |             ^^^^^^^^^^^^^^^^^
    |
-   = note: an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
+help: consider using a box or reference as appropriate
+  --> $DIR/issue-73886.rs:4:13
+   |
+LL |     let _ = 7u32 as Option<_>;
+   |             ^^^^
```

Comment:
This is a regression, but seeing the way _other_ error improve, I wouldn't want to block on it.